src/applications/disability-benefits/all-claims/tests/utils/clean-toxic-exposure-data.unit.spec.jsx

```diff
@@ -0,0 +1,1466 @@
+        'asbestos',
+      );
+      expect(result.toxicExposure.otherExposuresDetails).to.not.have.property(
+        'radiation',
+      );
+    });
+
+    it('should remove otherExposuresDetails for unselected exposures', () => {
+      const formData = {
+        toxicExposure: {
+          conditions: { cancer: true },
+          otherExposures: {
+            burnpits: false,
+            chemicals: true,
+          },
+          otherExposuresDetails: {
+            burnpits: {
+              startDate: '2003-01-01',
+              endDate: '2004-01-01',
+            },
+            chemicals: {
+              startDate: '2005-01-01',
+              endDate: '2006-01-01',
+            },
+          },
+        },
+      };
+
+      const result = cleanToxicExposureData(formData);
+
+      expect(result.toxicExposure.otherExposuresDetails).to.not.have.property(
+        'burnpits',
+      );
+      expect(result.toxicExposure.otherExposuresDetails).to.have.property(
+        'chemicals',
+      );
+    });
+
+    it('should remove specifyOtherExposures when empty string', () => {
+      const formData = {
+        toxicExposure: {
+          conditions: { asthma: true },
+          specifyOtherExposures: '   ',
+        },
+      };
+
+      const result = cleanToxicExposureData(formData);
+
+      expect(result.toxicExposure).to.not.have.property(
+        'specifyOtherExposures',
+      );
+    });
+
+    it('should keep specifyOtherExposures when string has content', () => {
+      const formData = {
+        toxicExposure: {
+          conditions: { asthma: true },
+          specifyOtherExposures: 'Mercury exposure',
+        },
+      };
+
+      const result = cleanToxicExposureData(formData);
+
+      expect(result.toxicExposure.specifyOtherExposures).to.equal(
+        'Mercury exposure',
+      );
+    });
+
+    it('should remove specifyOtherExposures when object description is empty', () => {
+      const formData = {
+        toxicExposure: {
+          conditions: { asthma: true },
+          specifyOtherExposures: {
+            description: '   ',
+            otherField: 'value',
+          },
+        },
+      };
+
+      const result = cleanToxicExposureData(formData);
+
+      expect(result.toxicExposure).to.not.have.property(
+        'specifyOtherExposures',
+      );
+    });
+
+    it('should keep specifyOtherExposures when object description has content', () => {
+      const formData = {
+        toxicExposure: {
+          conditions: { asthma: true },
+          specifyOtherExposures: {
+            description: 'Lead paint exposure',
+            otherField: 'value',
+          },
+        },
+      };
+
+      const result = cleanToxicExposureData(formData);
+
+      expect(result.toxicExposure.specifyOtherExposures).to.deep.equal({
+        description: 'Lead paint exposure',
+        otherField: 'value',
+      });
+    });
+
+    it('should remove multiple false conditions', () => {
```

Comment:
I wonder if **false** conditions shouldn't be removed as their presence in the submitted `InProgressForm` data helped to inform the existence of some invalid dates in the first place.